### PR TITLE
chore(llmobs): default role if not provided on the message

### PR DESCRIPTION
**Pull request by sabrenner**
Defaults the role on the messages in LLM input/output messages. This is a non-user facing change. These messages will still show up as the `Message` role in the UI - this change makes querying these fields more consistent.

MLOB-2688

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/llmobs/_utils.py [hunk 1]
```diff
@@ -224,6 +224,17 @@ def add_span_link(span: Span, span_id: str, trace_id: str, from_io: str, to_io:
     span._set_ctx_item(SPAN_LINKS, current_span_links)
 
 
+def enforce_message_role(messages: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    """
+    Enforce the message role for the LLM span event.
+    A message with a role of empty content will render as "Message" in the UI.
+    """
+    for message in messages:
+        if message.get("role") is None:
```
Comment (Yun-Kim) [inline ddtrace/llmobs/_utils.py:233]: Is this just to ensure there exists a role field?

File: ddtrace/llmobs/_utils.py [hunk 2]
```diff
@@ -224,6 +224,17 @@ def add_span_link(span: Span, span_id: str, trace_id: str, from_io: str, to_io:
     span._set_ctx_item(SPAN_LINKS, current_span_links)
 
 
+def enforce_message_role(messages: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    """
+    Enforce the message role for the LLM span event.
+    A message with a role of empty content will render as "Message" in the UI.
```
Comment (Yun-Kim) [inline ddtrace/llmobs/_utils.py:230]:
```suggestion
    Enforce that each message includes a role (empty "" by default) field.
```

File: ddtrace/llmobs/_utils.py [hunk 3]
```diff
@@ -224,6 +224,17 @@ def add_span_link(span: Span, span_id: str, trace_id: str, from_io: str, to_io:
...
+    """
+    for message in messages:
+        if message.get("role") is None:
+            message["role"] = ""
+    return messages
```
Comment (tillwf) [inline ddtrace/llmobs/_utils.py:235]: Can the value be `None`? If not, there is this nice [function](https://docs.vultr.com/python/standard-library/dict/setdefault) in Python which set to a default value if the key does not exists:

```python
for message in messages:
    message.setdefault("role", "")
return messages
```

Example:

```python
>>> msg = {"content": "Hey"}
>>> msg.setdefault("role", "")
''
>>> msg
{'content': 'Hey', 'role': ''}

>>> msg = {"content": "Hey", "role": "assistant"}
>>> msg.setdefault("role", "")
'assistant'
>>> msg
{'content': 'Hey', 'role': 'assistant'}
```

File: ddtrace/llmobs/_utils.py (hunk 1 above)
Comment (sabrenner) [reply ddtrace/llmobs/_utils.py:233]: yeah - will modify to `message.setdefault("role", "")` for a bit more simplicity

File: tests/llmobs/_utils.py [hunk 4]
```diff
@@ -101,8 +101,24 @@ def _expected_llmobs_llm_span_event(
     meta_dict = {"input": {}, "output": {}}
```
Comment (Kyle-Verhoog) [inline tests/llmobs/_utils.py:101]: can we add some additional test cases to ensure that the message is normalized if no role is specified or if it is set to `None` by the user?

**Review by Kyle-Verhoog — APPROVED** [review]
just a test case suggestion, otherwise lgtm

File: tests/llmobs/_utils.py (hunk 4 above)
Comment (sabrenner) [reply tests/llmobs/_utils.py:101]: added!

File: ddtrace/llmobs/_utils.py (hunk 1 above)
Comment (sabrenner) [reply ddtrace/llmobs/_utils.py]: was there any other concern with this block here?

Comment (Yun-Kim) [reply ddtrace/llmobs/_utils.py]: Nope!
